src/micromamba/config.cpp

```diff
@@ -194,194 +178,150 @@ set_config_path_command(CLI::App* subcom)
-        fs::path rc_source = env::expand_user(env::home_directory() / ".condarc");
-        std::ofstream rc_file;
-
-        std::string append_key = append_map.value().front();
-        std::string append_value = append_map.value().back();
+    fs::path rc_source = env::expand_user(env::home_directory() / ".condarc");
 
-        if (file_path.configured())
-        {
-            path::touch(file_path.value().string(), true);
-            rc_source = env::expand_user(file_path.value()).string();
-        }
-        else if (env_path.configured())
-        {
-            std::string path = fs::path(ctx.target_prefix / ".condarc");
-            env_path_setup(path, rc_source);
-        }
-        else if (system_path.configured())
-        {
-            system_path_setup(rc_source);
-        }
+    if (file_path.configured())
+    {
+        path::touch(file_path.value().string(), true);
+        rc_source = env::expand_user(file_path.value()).string();
+    }
+    else if (env_path.configured())
+    {
+        std::string path = fs::path(ctx.target_prefix / ".condarc");
+        env_path_setup(path, rc_source);
+    }
+    else if (system_path.configured())
+    {
+        system_path_setup(rc_source);
+    }
 
-        YAML::Node rc_YAML = YAML::LoadFile(rc_source.string());
+    YAML::Node node = YAML::LoadFile(rc_source.string());
+    for (auto& pair : specs)
+        set_sequence_to_yaml(node, pair.first, pair.second, opt);
 
-        // look for append key in YAML
-        for (auto v : rc_YAML)
-        {
-            if (v.first.as<std::string>() == append_key)
-            {
-                // if key was found, look for the same value
-                for (std::size_t j = 0; j < rc_YAML[append_key].size(); j++)
-                {
-                    if (rc_YAML[append_key][j].as<std::string>() == append_value)
-                    {
-                        // if value exists, remove it so it can be appended to end of the list
-                        // later on
-                        rc_YAML[append_key].remove(j);
-                        break;
-                    }
-                }
-            }
-        }
+    std::ofstream rc_file;
+    rc_file.open(rc_source.string(), std::ofstream::in | std::ofstream::trunc);
+    rc_file << node << std::endl;
 
-        if (is_key_valid(append_key) && append_map.value().size() < 3)
-        {
-            rc_YAML[append_key].push_back(append_value);
-        }
-        else
-        {
-            std::cout
-                << "Append key is invalid or it's not present in file or more than one key was received"
-                << std::endl;
-        }
+    config.operation_teardown();
+}
 
-        // if the rc file is being modified, it's necessary to rewrite it
-        rc_file.open(rc_source.string(), std::ofstream::in | std::ofstream::trunc);
-        rc_file << rc_YAML << std::endl;
+void
+set_config_prepend_command(CLI::App* subcom)
+{
+    set_config_sequence_command(subcom);
```

Comment:
Maybe here we add the specific description.

Reply:
yep :+1:, see my comment above

Reply:
Yeah, I saw it was overwritten, was just thinking about code clarity :)